File: Cantera/matlab/cantera/examples/rankine.m
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% 
%
%  An ideal Rankine cycle.
%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


% create an object representing water
w = Water;

% start with saturated liquid water at 300 K
set(w,'T',300.0,'Sat','Liquid');
h1 = enthalpy_mass(w);
s1 = entropy_mass(w);
p1 = pressure(w);

% pump it isentropically to 10 MPa
set(w,'S',s1,'P',1.0e7);
h2 = enthalpy_mass(w);
p2 = pressure(w);

pump_work = h2 - h1;


% heat to 1500 K at constant pressure
set(w,'T',1500.0,'P',p2);
h3 = enthalpy_mass(w);
s3 = entropy_mass(w);

heat_added = h3 - h2;


% expand isentropically to the initial pressure
set(w,'S',s3,'P',p1);
h4 = enthalpy_mass(w);
x4 = vaporFraction(w);

work_output = h3 - h4;

% compute the efficiency
efficiency = (work_output - pump_work)/heat_added

File: Cantera/python/examples/rankine.py
#
# an ideal Rankine cycle
#
from Cantera import *
from Cantera.pureFluids import Water

w = Water()

# start with saturated liquid water at 300 K
w.setTemperature(300.0)
w.setState_satLiquid()
h1 = w.enthalpy_mass()
p1 = w.pressure()

# pump it isentropically to 10 MPa
w.setState_SP(w.entropy_mass(), 1.0e7)
h2 = w.enthalpy_mass()

pump_work = h2 - h1

# heat at constant pressure to 1500 K
w.setState_TP(1500.0, w.pressure())
h3 = w.enthalpy_mass()

heat_in = h3 - h2

# expand isentropically back to 300 K
w.setState_SP(w.entropy_mass(), p1)
h4 = w.enthalpy_mass()

work_out = h3 - h4
heat_out = h4 - h1

efficiency = (work_out - pump_work)/heat_in

print 'efficiency = ',efficiency

